cli/popper/scm.py

```diff
@@ -15,14 +15,23 @@ def init_repo_object():
     try:
         repo = git.Repo(search_parent_directories=True)
     except git.exc.InvalidGitRepositoryError:
-        log.fail('Unable to find root of a Git repository.')
+        log.warn('Unable to find root of a Git repository.')
+
+    if repo:
+        if repo.head.is_detached:
+            log.fail('HEAD is in detached state.')
```

Comment:
this would impede popper to run at all on any repo in detached `HEAD` state, right? Instead, we could move this check to another function (e.g. `check_if_detached_head`) that is invoked only when `CI` set is enabled (from `popper run`).